repose-aggregator/components/services/datastore-service/datastore-service-impl/src/main/java/org/openrepose/core/services/datastore/impl/remote/RemoteDatastore.java

```diff
@@ -0,0 +1,244 @@
+
+import org.openrepose.commons.utils.encoding.EncodingProvider;
+import org.openrepose.core.services.datastore.Datastore;
+import org.openrepose.core.services.datastore.DatastoreOperationException;
+import org.openrepose.core.services.datastore.Patch;
+import org.openrepose.core.services.datastore.distributed.DistributedDatastore;
+import org.openrepose.core.services.datastore.distributed.RemoteBehavior;
+import org.openrepose.core.services.datastore.distributed.SerializablePatch;
+import org.openrepose.core.services.datastore.hash.MessageDigestFactory;
+import org.openrepose.core.services.datastore.impl.distributed.DatastoreAction;
+import org.openrepose.core.services.datastore.impl.distributed.remote.RemoteCommandExecutor;
+import org.openrepose.core.services.datastore.impl.distributed.remote.RemoteConnectionException;
+import org.openrepose.core.services.datastore.impl.distributed.remote.command.Delete;
+import org.openrepose.core.services.datastore.impl.distributed.remote.command.Get;
+import org.openrepose.core.services.datastore.impl.distributed.remote.command.Put;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.TimeUnit;
+
+public class RemoteDatastore implements DistributedDatastore {
+
+    public static final String DATASTORE_NAME = "distributed/remote";
+    private static final Logger LOG = LoggerFactory.getLogger(RemoteDatastore.class);
+    private static final int DEFAULT_TTL = 5;
+
+    private final Datastore localDatastore;
+    private final RemoteCommandExecutor remoteCommandExecutor;
+    private final EncodingProvider encodingProvider;
+    private final MessageDigestFactory hashProvider;
+    private final String datasetPrefix;
+    private final String name;
+    private final InetSocketAddress target;
+    private final String connPoolId;
+    private final boolean useHttps;
+
+    public RemoteDatastore(RemoteCommandExecutor remoteCommandExecutor,
+                           String datastorePrefix, Datastore localDatastore, MessageDigestFactory hashProvider,
+                           EncodingProvider encodingProvider, InetSocketAddress target, String connPoolId, boolean useHttps) {
+
+        this.name = DATASTORE_NAME;
+        this.encodingProvider = encodingProvider;
+        this.hashProvider = hashProvider;
+        this.datasetPrefix = datastorePrefix;
+        this.target = target;
+        this.connPoolId = connPoolId;
+        this.useHttps = useHttps;
+
+        this.localDatastore = localDatastore;
+        this.remoteCommandExecutor = remoteCommandExecutor;
+        if (localDatastore == null) {
+            LOG.warn("Local datastore is null");
+        }
+    }
+
+    private Object performAction(String name, DatastoreAction action) {
+        return performRemote(name, action);
+    }
```

Comment:
Since nothing is ever being performed locally, I'll keep the `performRemote` name.